plugins/terraform/terraform.go

```diff
@@ -0,0 +1,20 @@
+package terraform
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+)
+
+func TerraformCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Terraform CLI",
+		Runs:    []string{"terraform"},
+		DocsURL: sdk.URL("https://developer.hashicorp.com/terraform/cli"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
```

Comment:
`terraform login` and `terraform logout`?

Reply:
Since this PR is mostly about adding support for plugins that support multiple credential types, and Terraform is just an example, I will create separate issues for addressing Terraform specific improvements. Logging in and out of terraform is not absolutely required for using terraform with aws for example, as authentication is done at your AWS account level.

Reply:
Good point, will do!

Reply:
Done!